spec-main/api-net-log-spec.js

```diff
@@ -13,7 +13,7 @@ const dumpFileDynamic = path.join(os.tmpdir(), 'net_log_dynamic.json')
 const { expect } = chai
 chai.use(dirtyChai)
 const isCI = global.isCI
-const netLog = session.fromPartition('net-log').netLog
+const testNetLog = () => session.fromPartition('net-log').netLog
```

Comment:
What does this change have to do with this PR?

Reply:
It's relevant because this PR alters session construction logic, and eagerly creating the `net-log` session during testing was preventing me from filtering tests properly.

This code gets run when the tests launch regardless of whether the netLog tests are filtered in by `-g`. The upshot was that when this code was causing in-memory sessions to crash on initialization, i couldn't test the rest of the code that didn't rely on in-memory sessions.